src/stx/stl/trex_stl_dp_core.cpp

```diff
@@ -1462,9 +1462,15 @@ TrexStatelessDpCore::add_timesync_node(PerPortProfile *profile,
     node->m_next_time_offset = SYNC_TIME_OUT;
     node->init();
 
-    rte_mbuf_t *m = CGlobalInfo::pktmbuf_alloc_local(node->get_socket_id(), (ETH_HDR_LEN + PTP_DELAYRESP_LEN));
-    assert(m);
-    node->m = m;
+    node->m = nullptr;
+
+    if (CGlobalInfo::m_options.is_timesync_L2()) {
+        node->m = CGlobalInfo::pktmbuf_alloc_local(node->get_socket_id(), (ETH_HDR_LEN + PTP_DELAYRESP_LEN));
+    } else {
+        node->m = CGlobalInfo::pktmbuf_alloc_local(node->get_socket_id(), (ETH_HDR_LEN + IPV4_HDR_LEN + UDP_HEADER_LEN + PTP_DELAYRESP_LEN));
+    }
```

Comment:
Sorry, but there is too much code IMO.  First L1465 is completely redundant.  Second why not a one-liner like this:

```
node->m = CGlobalInfo::pktmbuf_alloc_local(node->get_socket_id(),
    ETH_HDR_LEN + PTP_DELAYRESP_LEN +
    (CGlobalInfo::m_options.is_timesync_L2() ? 0 : (IPV4_HDR_LEN + UDP_HEADER_LEN)));
```

Reply:
IMHO normal if is cleaner here